### PR TITLE
Add WebGL support

### PR DESCRIPTION
**Pull request by jorolf**
pr'ing because I want to know if this is the right direction and wether I should continue (adding the remaining WebGL functions is going to take a while). 

I'm also unsure if I missed any naming rules, Resharper didn't like this solution.

### CODE REVIEW

**Comment by galvesribeiro** [comment]
Hello @jorolf! Thanks for the PR! Outstanding work!

Please keep going, you are on the right track! 

> Resharper didn't like this solution.

Yeah, Resharper don't like when solutions have custom MSBuild targets. I would say, drop Resharper if you are using VS2017+ 🔥 

Please let me know if you have any problems.

**Comment by jorolf** [comment]
It's mostly code style warnings that resharper doesn't like. Stuff like "ConstString" instead of "CONST_STRING", those rules should be easy to add. Should I create another pull request for those?

**Comment by galvesribeiro** [comment]
> It's mostly code style warnings that resharper doesn't like. Stuff like "ConstString" instead of "CONST_STRING", those rules should be easy to add. Should I create another pull request for those?

I would rather prefer to keep those as is for now as it is spread across other projects.

File: Canvas.sln.DotSettings [hunk 1]
```diff
@@ -0,0 +1,7 @@
+﻿<wpf:ResourceDictionary xml:space="preserve" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" xmlns:s="clr-namespace:System;assembly=mscorlib" xmlns:ss="urn:shemas-jetbrains-com:settings-storage-xaml" xmlns:wpf="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
```
Comment (galvesribeiro) [inline Canvas.sln.DotSettings]: > That's what I mean. I would add rules to tell resharper that the current style is the correct one

Sorry but I don't meant that we should add to the repo that file `Canvas.sln.DotSettings`.

We don't want to have 3rd party tooling-specific config settings on the repo. 

Can you please undo that?

**Comment by jorolf** [comment]
For some reason every class was in the ``Blazor.Extensions`` namespace instead of ``Blazor.Extensions.Canvas``. Was that intented?

**Comment by galvesribeiro** [comment]
> For some reason every class was in the `Blazor.Extensions` namespace instead of `Blazor.Extensions.Canvas`. Was that intented?

Yes. All the extensions are under `Blazor.Extensions` like this one https://github.com/BlazorExtensions/SignalR/blob/master/src/Blazor.Extensions.SignalR/HubConnectionBuilder.cs#L3

**Comment by jorolf** [comment]
Hmm... weird. The projects I've had worked with so far enforced that the namespace should be equal to the path. What are the reasons behind this?

**Comment by galvesribeiro** [comment]
> Hmm... weird. The projects I've had worked with so far enforced that the namespace should be equal to the path. What are the reasons behind this?

Yeah, I would agree that a general idea would be to follow that pattern.

We thought on Blazor Extensions initially to be a single assembly.  Then we thought it would be way more complex to manage it then we decide to add one assembly por group of APIs but that each one would "enhance" the core namespace with its types. That way, both the JS side would have a single root'ed object and the C# side would follow the root namespace. 

In most of cases, the components here have very few public types so its ok to be on the root namespace.

**Comment by galvesribeiro** [comment]
Btw, lets keep the PR discussion focused on the code review. Easier for others to read. I'm glad to chat about anything else related to the extensions on the Gitter lobby. Feel free to ping me there.

**Pull request merged**